src/container.ts

```diff
@@ -918,6 +918,31 @@ export class Container {
 			},
 		});
 	}
+
+	@memoize()
+	private get baseGkDevUri(): Uri {
+		if (this.env === 'staging') {
+			return Uri.parse('https://staging.gitkraken.dev');
+		}
+
+		if (this.env === 'dev') {
+			return Uri.parse('https://dev.gitkraken.dev');
+		}
+
+		return Uri.parse('https://gitkraken.dev');
+	}
+
+	getGkDevUri(path?: string, query?: string): Uri {
+		let uri = path != null ? Uri.joinPath(this.baseGkDevUri, path) : this.baseGkDevUri;
+		if (query != null) {
+			uri = uri.with({ query: query });
+		}
+		return uri;
+	}
+
+	generateWebGkDevUrl(path?: string): string {
+		return this.getGkDevUri(path, '?source=gitlens').toString();
+	}
```

Comment:
Do we need this? I'm not sure `source=gitlens` is something gk.dev can support on any URL, but if so, it seems like that should just be integrated into `getGKDevUri`

@schinizel can we add `source=gitlens` to any gk.dev route?

Reply:
I would move the `source=gitlens` part directly into `getGkDevUri`, as it seems based on responses from the team, that this would never break functionality on GKDev pages. Then we can get rid of this.

Reply:
https://gitkraken-hq.slack.com/archives/C05H8JUFK8Q/p1718141676567519

... while it won't break anything, why add it to every route/request? I will post in the Slack thread.